### DISCUSSION TITLE
Handing rows deleted at server in a offline scenario.

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#347] [Q&A] [answered]
In my setup (which maybe incorrect), if I add a todo item at the client (offline) then sync, it appears at the server... so far so good. If I then delete the row at the server and sync the client, the row is still there.

Is this expected behaviour ? How can this be managed ?

**Comment by fileman** [comment]
With offline table you should set [EnableSoftDelete](https://docs.microsoft.com/en-us/azure/developer/mobile-apps/azure-mobile-apps/howto/server/dotnet-core#configure-table-controller-options) to true in controller options, and then use PurgeItem on client, those are my working implementations:

- In controller ctor: `Options.EnableSoftDelete = true;`
- In client: `await _offlineTable.PullItemsAsync();
            var purgeDeletedItemsQuery = _offlineTable.CreateQuery().Where(x => x.Deleted == true);
            await _offlineTable.PurgeItemsAsync(purgeDeletedItemsQuery, new PurgeOptions { DiscardPendingOperations = false });`

Maybe @adrianhall can give you a better reply and improve docs and sample with this use case.

**Reply by henda79** [reply]
Thanks for the reply, but I can't see how this helps or maybe I misunderstand.

I am talking about deleting the row from the server side using something like SMSS, then refreshing the client. The client still holds the old row and does not notice it has been deleted.

**Comment by adrianhall** [comment, marked as answer]
What you are seeing is "expected behavior".  The client doesn't know that the row has been deleted, so it sticks around.

Instead of deleting the row:

1. Enable soft delete on the table.
2. Mark the row as deleted (set Deleted = true, change version, updated at should be auto updated, but if it isn't do that too)
3. Your client will then automatically mark the row as deleted and you can purge them.

Another alternative, if you can't enable soft delete:

1. Purge all items from the table
2. Re-pull

**Reply by henda79** [reply]
Thanks for clearing that up. I am used to working with the old SQL Merge Replication and if we deleted a row directly in the database on the server it would also get deleted on all the clients that subscribed to the database.

I'll have a play with the soft delete option.
